File: 2st/Riyeon/추억 점수.java
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        for(int i=0; i<photo.length; i++){
            int cnt=0;
            for(int j=0; j<photo[i].length; j++)
                for(int n=0; n<name.length; n++){
                    if(photo[i][j].equals(name[n])){
                        cnt += yearning[n];
                        break;
                    }
                }
            answer[i] = cnt;
        }
        return answer;
    }
}

File: 2st/Riyeon/추억 점수.py
def solution(name, yearning, photo):
    di = dict()
    for i in range(len(name)):
        di[name[i]] = yearning[i]
    
    answer = []
    for li in photo:
        num=0
        for i in li:
            if i in name:
                num += di[i]
        answer.append(num)
    return answer